photon-client/src/main/scala/com/linkedin/photon/ml/io/scopt/ScoptParserReads.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.io.scopt
+
+import org.apache.hadoop.fs.Path
+
+import com.linkedin.photon.ml.{DataValidationType, HyperparameterTuningMode, TaskType}
+import com.linkedin.photon.ml.DataValidationType.DataValidationType
+import com.linkedin.photon.ml.HyperparameterTuningMode.HyperparameterTuningMode
+import com.linkedin.photon.ml.TaskType.TaskType
+import com.linkedin.photon.ml.evaluation.EvaluatorType
+import com.linkedin.photon.ml.io.ModelOutputMode
+import com.linkedin.photon.ml.io.ModelOutputMode.ModelOutputMode
+import com.linkedin.photon.ml.normalization.NormalizationType
+import com.linkedin.photon.ml.normalization.NormalizationType.NormalizationType
+import com.linkedin.photon.ml.util.{DateRange, DaysRange, Utils}
+
+/**
+ * Implicit Scopt [[scopt.Read]] objects, used to read Scopt parameters directly into additional non-string types.
+ */
+object ScoptParserReads {
```

Comment:
@joshvfleming 
@fastier-li 
@Adama94

Implicit `scopt.Read` objects for parsing parameters of non-trivial types directly from `String` using Scopt.